### PR TITLE
O3-999: Added outpatient app side panel search bar

### PR DESCRIPTION
**Pull request by CynthiaKamau**
## Requirements

- [x] This PR has a title that briefly describes the work done, on ticket O3-999.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Screenshots
Patient Queue Add Patient To List

## Issue

https://issues.openmrs.org/projects/O3/issues/O3-999

### CODE REVIEW

File: packages/esm-outpatient-app/src/patient-search-icon/patient-search-icon.component.scss [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  background-color: $inverse-02;
+}
+
+
+
+
+
+ 
+
+
```
Comment (anjula-sack) [inline packages/esm-outpatient-app/src/patient-search-icon/patient-search-icon.component.scss:69]: Can we remove these extra lines?

File: packages/esm-outpatient-app/src/overlay.scss [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+.buttonsGroup button {
+  max-width: unset !important;
+}
+
+:global(.omrs-breakpoint-lt-desktop) .buttonsGroup {
+  padding: $spacing-06 $spacing-05;
+  background-color: $ui-02;
+}
```
Comment (anjula-sack) [inline packages/esm-outpatient-app/src/overlay.scss:71]: End of line is needed to be added

File: packages/esm-outpatient-app/src/patient-search-icon/patient-search-icon.component.scss [hunk 3]
```diff
@@ -0,0 +1,69 @@
+@import "~@openmrs/esm-styleguide/src/vars";
+@import "~carbon-components/src/globals/scss/vendor/@carbon/layout/scss/generated/spacing";
+$inverse-02:#393939;
```
Comment (brandones) [inline packages/esm-outpatient-app/src/patient-search-icon/patient-search-icon.component.scss:3]: This is only used once. Just inline it. If you believe it is missing from [esm-styleguide](https://github.com/openmrs/openmrs-esm-core/blob/master/packages/framework/esm-styleguide/src/_vars.scss), please add it there.

File: packages/esm-outpatient-app/src/overlay.scss [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  bottom: 0;
+  left: 0;
+  right: 0;
+  z-index: 9999;
+  background-color: $ui-01;
+  overflow: hidden;
+  padding-top: $spacing-09;
+  display: grid;
+  grid-template-rows: 1fr auto;
+}
```
Comment (brandones) [inline packages/esm-outpatient-app/src/overlay.scss:26]: @denniskigen The amount of CSS/SCSS in our project makes me uneasy. Is this doing anything special? Is there a way we can standardize some of this using esm-styleguide? Ideally this PR should involve almost no custom styling.

(I'm not going to block this PR on us getting our styling situation figured out, of course, but I want to start the conversation).

File: packages/esm-outpatient-app/src/home.component.tsx [hunk 5]
```diff
@@ -1,15 +1,79 @@
...
+                <Tile>
+                  <h2> Active Visits</h2>
+                </Tile>
+              </Column>
+
+              <Column sm={12} md={4} lg={4}>
+                <Tile>
+                  <ContentSwitcher>
+                    <Switch name={'first'} text="Default" />
+                    <Switch name={'second'} text="Large" />
```
Comment (brandones) [inline packages/esm-outpatient-app/src/home.component.tsx:50]:
```suggestion
                    <Switch name={'default'} text="Default" />
                    <Switch name={'large'} text="Large" />
```

File: packages/esm-outpatient-app/src/home.component.tsx [hunk 6]
```diff
@@ -1,15 +1,79 @@
...
+}
+
+interface AdvancedPatientListState {
+  type: RouteStateTypes.ADVANCED_PATIENT_LIST;
+}
+
+type RouteState = AddPatientRouteState | AdvancedPatientListState;
+
 const Home: React.FC<HomeProps> = () => {
+  const [routeState, setRouteState] = useState<RouteState>({ type: RouteStateTypes.ADD_PATIENT_LIST });
```
Comment (brandones) [inline packages/esm-outpatient-app/src/home.component.tsx:28]: Why not just 

```suggestion
  const [routeState, setRouteState] = useState<RouteState>(RouteState.ADD_PATIENT_LIST);
```

and simplify the types accordingly?

**Comment by brandones** [comment]
Is there a ticket for this change? There are things about it that I don't really understand. Like what's "advanced patient list."

**Comment by CynthiaKamau** [comment]
> Is there a ticket for this change? There are things about it that I don't really understand. Like what's "advanced patient list."

https://issues.openmrs.org/projects/O3/issues/O3-999

**Comment by brandones** [comment]
Thanks @CynthiaKamau . Could you add screenshots or a gif showing the other aspects of the UI? e.g. the button for advanced search, and the advanced search pane?

**Comment by CynthiaKamau** [comment]
> Thanks @CynthiaKamau . Could you add screenshots or a gif showing the other aspects of the UI? e.g. the button for advanced search, and the advanced search pane?

Yes i will

**Pull request closed without merging**